File: app/src/main/java/com/investmentassistant/MainActivity.java
package com.investmentassistant;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private EditText etPrincipal;
    private EditText etRate;
    private EditText etYears;
    private TextView tvResult;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        etPrincipal = findViewById(R.id.et_principal);
        etRate      = findViewById(R.id.et_rate);
        etYears     = findViewById(R.id.et_years);
        tvResult    = findViewById(R.id.tv_result);

        Button btnCalculate = findViewById(R.id.btn_calculate);
        btnCalculate.setOnClickListener(v -> calculate());
    }

    private void calculate() {
        String principalStr = etPrincipal.getText().toString().trim();
        String rateStr      = etRate.getText().toString().trim();
        String yearsStr     = etYears.getText().toString().trim();

        if (principalStr.isEmpty() || rateStr.isEmpty() || yearsStr.isEmpty()) {
            tvResult.setText(R.string.error_empty_fields);
            return;
        }

        double principal = Double.parseDouble(principalStr);
        double rate      = Double.parseDouble(rateStr) / 100.0;
        int    years     = Integer.parseInt(yearsStr);

        double amount = principal * Math.pow(1 + rate, years);
        double profit = amount - principal;

        tvResult.setText(getString(R.string.result_format,
                String.format("%.2f", amount),
                String.format("%.2f", profit)));
    }
}

File: app/src/main/kotlin/com/rahulsharmafca199/investmentassistant/MainActivity.kt
package com.rahulsharmafca199.investmentassistant

import android.os.Bundle
import android.view.View
import android.widget.Toast
import kotlin.math.pow
import androidx.appcompat.app.AppCompatActivity
import com.rahulsharmafca199.investmentassistant.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        setSupportActionBar(binding.toolbar)

        setupInvestmentCalculator()
    }

    private fun setupInvestmentCalculator() {
        binding.btnCalculate.setOnClickListener {
            val principalText = binding.etPrincipal.text.toString()
            val rateText = binding.etRate.text.toString()
            val yearsText = binding.etYears.text.toString()

            if (principalText.isEmpty() || rateText.isEmpty() || yearsText.isEmpty()) {
                Toast.makeText(this, "Please fill in all fields", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            val principal = principalText.toDoubleOrNull()
            val rate = rateText.toDoubleOrNull()
            val years = yearsText.toIntOrNull()

            if (principal == null || rate == null || years == null) {
                Toast.makeText(this, "Please enter valid numbers", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            val finalAmount = calculateCompoundInterest(principal, rate / 100, years)
            val totalInterest = finalAmount - principal

            binding.tvResult.visibility = View.VISIBLE
            binding.tvResult.text = String.format(
                "Final Amount: ₹%.2f\nTotal Interest: ₹%.2f",
                finalAmount,
                totalInterest
            )
        }

        binding.btnClear.setOnClickListener {
            binding.etPrincipal.text?.clear()
            binding.etRate.text?.clear()
            binding.etYears.text?.clear()
            binding.tvResult.visibility = View.GONE
        }
    }

    private fun calculateCompoundInterest(principal: Double, rate: Double, years: Int): Double {
        return principal * (1 + rate).pow(years.toDouble())
    }
}
